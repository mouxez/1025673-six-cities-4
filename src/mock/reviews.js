const reviews = [
  {
    "id": 1,
    "avatarUrl": `http://picsum.photos/90/90?r=1`,
    "user": `John`,
    "rating": 3.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `December 24, 2018`,
  },
  {
    "id": 2,
    "avatarUrl": `http://picsum.photos/90/90?r=2`,
    "user": `Ivan`,
    "rating": 4.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `December 20, 2018`,
  },
  {
    "id": 3,
    "avatarUrl": `http://picsum.photos/90/90?r=3`,
    "user": `Sveta`,
    "rating": 1.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `July 24, 2018`,
  },
  {
    "id": 4,
    "avatarUrl": `http://picsum.photos/90/90?r=4`,
    "user": `Elly`,
    "rating": 3.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `May 14, 2018`,
  },
  {
    "id": 5,
    "avatarUrl": `http://picsum.photos/90/90?r=5`,
    "user": `Kate`,
    "rating": 0.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `December 21, 2019`,
  },
  {
    "id": 6,
    "avatarUrl": `http://picsum.photos/90/90?r=6`,
    "user": `Mike`,
    "rating": 2.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `March 11, 2020`,
  },
  {
    "id": 7,
    "avatarUrl": `http://picsum.photos/90/90?r=7`,
    "user": `Jay`,
    "rating": 5.0,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `October 14, 2016`,
  },
  {
    "id": 8,
    "avatarUrl": `http://picsum.photos/90/90?r=8`,
    "user": `Sergey`,
    "rating": 4.7,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `May 22, 2010`,
  },
  {
    "id": 9,
    "avatarUrl": `http://picsum.photos/90/90?r=9`,
    "user": `Elena`,
    "rating": 2.7,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `November 11, 2019`,
  },
  {
    "id": 10,
    "avatarUrl": `http://picsum.photos/90/90?r=10`,
    "user": `Tema`,
    "rating": 1.7,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `June 07, 2012`,
  },
  {
    "id": 11,
    "avatarUrl": `http://picsum.photos/90/90?r=11`,
    "user": `Olga`,
    "rating": 4.9,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `July 01, 2020`,
  }
];

export {reviews};
