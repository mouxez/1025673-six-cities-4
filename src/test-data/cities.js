const cities = [{location: [59.3198, 18.0601], name: `Stockholm`}];

export {cities};
