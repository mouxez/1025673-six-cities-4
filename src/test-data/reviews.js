const reviews = [
  {
    "avatarUrl": `http://picsum.photos/90/90?r=1`,
    "user": `John`,
    "rating": 3.3,
    "description": `A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
    "date": `2020-07-02`,
  }
];

export {reviews};
