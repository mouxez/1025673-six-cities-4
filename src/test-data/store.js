import configureStore from 'redux-mock-store';
import {cities} from './cities';
import {offers} from './offers';

const mockStore = configureStore([]);

const store = mockStore({
  activeCityName: cities[0].name,
  offers,
  cities,
});

export {store};
