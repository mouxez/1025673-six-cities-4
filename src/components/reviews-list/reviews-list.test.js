import React from 'react';
import renderer from 'react-test-renderer';
import ReviewsList from './reviews-list';
import {reviews} from '../../test-data/reviews';

test(`ReviewsList should render correctly`, () => {
  const tree = renderer.create(
      <ReviewsList reviews={reviews}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
