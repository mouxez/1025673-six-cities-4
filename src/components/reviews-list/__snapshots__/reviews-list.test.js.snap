// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList should render correctly 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="http://picsum.photos/90/90?r=1"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          John
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              3.3
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-07-02"
        >
          July 2020
        </time>
      </div>
    </li>
  </ul>,
]
`;
